SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -1070,23 +1070,14 @@ void USpatialNetDriver::NotifyActorDestroyed(AActor* ThisActor, bool IsSeamlessT
 		{
 			if (!World->bBegunPlay)
 			{
-				// When running in PIE, Blueprint loaded sub-levels can be duplicated and immediately unloaded.
-				// This causes each actor in the level to be destroyed, which we process here and if the actor is replicated etc., attempt
-				// to create a tombstone for. This would be incorrect behaviour, as the level and actors were never intentionally loaded,
-				// but are present just as an effect of the world duplication process.
-				// So we ignore this destroy call if the world hasn't begun play, and it was duplicated from PIE.
-				if (ThisActor->GetLevel() != nullptr && ThisActor->GetLevel()->bWasDuplicatedForPIE)
-				{
-					UE_LOG(LogSpatialOSNetDriver, Verbose,
-						   TEXT("USpatialNetDriver::NotifyActorDestroyed ignored as level was duplicated for PIE. Actor: %s."),
-						   *ThisActor->GetName());
-				}
-				else
-				{
-					UE_LOG(LogSpatialOSNetDriver, Error,
-						   TEXT("USpatialNetDriver::NotifyActorDestroyed ignored because world hasn't begun play. Actor: %s."),
-						   *ThisActor->GetName());
-				}
+				// PackageMap != nullptr implies the spatial connection is connected, however World::BeginPlay may not have been called yet
+				// which means we are still in a UEngine::LoadMap call. During the initial load process, actors are created and destroyed in
+				// the following scenarios:
+				// - When running in PIE, Blueprint loaded sub-levels can be duplicated and immediately unloaded.
+				// - ChildActorComponent::OnRegister
+				UE_LOG(LogSpatialOSNetDriver, Verbose,
+					   TEXT("USpatialNetDriver::NotifyActorDestroyed ignored because world hasn't begun play. Actor: %s."),
+					   *ThisActor->GetName());
```

Comment:
Kind of worried maybe we would be swallowing some useful errors, but I think this is quite rare.

Reply:
Yeah I agree, I wasn't sure all the paths through here since it's a pretty weird edgecase. Overall I think it's mostly ok because LoadMap calls World::beginplay, followed by the rest of the actors so I guess it's safe to assume that real code hasn't actually started yet? But I could be wrong if there are other edgecases, like server/client travel?

Reply:
(Merging for 0.13 urgency but if you think there's something else to investigate here we could follow up with more fixes)